libs/base/Data/SnocList.idr

```diff
@@ -96,6 +95,35 @@ public export
 Monoid (SnocList a) where
   neutral = Lin
 
+public export
+Foldable SnocList where
```

Comment:
This instance is inconsistent with the `Traversable` one.
* this insists that `SnocList` is just a normal list with a flipped `(::)`
* but traversable insists that `SnocList` is a reversed list